### PR TITLE
Sr labs audit ido pallet week2

### LINKED ISSUE #248: [IDO pallet] Storage items are never cleaned up which lead to storage bloating
https://github.com/Polkadex-Substrate/audit-sr-labs/issues/8

# Summary

In the IDO pallet (Polkadex/pallets/Polkadex_Ido/src/lib.rs) the extrinsics `show_interest_in_round` and `whitelist_investor` could exhaust the storage since they are never cleaned up and they're relatively cheap to call.

# Issue details

The storage item `InterestedParticipants` in the IDO pallet (Polkadex/pallets/Polkadex_Ido/src/lib.rs) stores a vector of interested participants and is defined as follows:

```rust
InterestedParticipants get(fn get_interested_particpants): map hasher(identity) T::Hash => Vec<T::AccountId>;

```

Any user can call the `show_interest_in_round` extrinsic and add a new item to this list which is never cleaned up, even after the funding round is ended. 

The same issue also exists for `whitelist_investor` where any user who has created a funding round can add new rows to the `WhiteListInvestors` hash map shown below:

```rust
WhiteListInvestors get(fn get_whitelist_investors): double_map hasher(identity) T::Hash, hasher(identity) T::AccountId  => T::Balance;

```

# Risk

Given that there are no storage deposits for these items, these extrinsics are relatively cheap to call and an attack could use them to fill up the blockchain storage. A full storage is problematic because it could lead to an infeasible amount of storage being required to run a blockchain node.

# Mitigation

We suggest adding a mechanism to clean up the storage after the funding round is ended.

### PR DESCRIPTION
**Pull request by mambisi**
Branched from #224 
Resolves #248

### CODE REVIEW

**Review by simonsso — CHANGES_REQUESTED** [review]
Please describe in the PR what is changed and why.

(it makes it easier for a reviewer to understand what we are looking at and what problem it is solving)

**Review by simonsso — CHANGES_REQUESTED** [review]
Where is this cleaning up storage bloat and what review comments is it solving?

File: pallets/Polkadex_Ido/src/lib.rs [hunk 1]
```diff
@@ -228,6 +228,21 @@ decl_module! {
 
         fn deposit_event() = default;
 
+        fn on_initialize(block_number: T::BlockNumber) -> Weight {
+
+            let call_weight: Weight = T::DbWeight::get().reads_writes(1, 1);
+            // Clean up WhiteListInvestors and InterestedParticipants in all expired rounds
```
Comment (mambisi) [inline pallets/Polkadex_Ido/src/lib.rs:234]: @simonsso  It cleans up expired funding rounds when a new block is created

**Comment by simonsso** [comment]
This branch is obsoleted

**Pull request closed without merging**